bot.py

```diff
@@ -700,11 +700,215 @@ def cmd_shoutout(match: Match[str]) -> Response:
     )
 
 
+class YouTubeVideo(NamedTuple):
+    video_id: str
+    title: str
+    playlist_id: str
+    playlist_position: int
+
+    def chat_message(self) -> str:
+        return (
+            f'{self.title} - '
+            f'https://www.youtube.com/watch?v={self.video_id}&list={self.playlist_id}&index={self.playlist_position}'  # noqa: E501
+        )
+
+
+async def ensure_youtube_playlist_videos_table_exists(
+    db: aiosqlite.Connection
+) -> None:
+    await db.execute(
+        'CREATE VIRTUAL TABLE IF NOT EXISTS youtube_playlist_videos using FTS5'
+        '('
+        '   playlist_id,'
+        '   title,'
+        '   video_id,'
+        '   playlist_position'
+        ')',
+    )
+    await db.commit()
+
+
+async def clear_youtube_playlist_from_store(
+    db: aiosqlite.Connection,
+    playlist_id: str
+) -> None:
+    await db.execute(
+        'DELETE FROM youtube_playlist_videos WHERE playlist_id = ?', (
+            playlist_id,
+        ),
+    )
+    await db.commit()
+
+
+async def fetch_youtube_playlist_videos(
+        playlist_id: str,
+        *,
+        youtube_api_key: str
+) -> List[YouTubeVideo]:
+    url = 'https://www.googleapis.com/youtube/v3/playlistItems'
+    params = {
+        'part': 'snippet',
+        'playlistId': playlist_id,
+        'key': youtube_api_key,
+        'maxResults': 50,
+    }
+    playlist_videos = []
+    more_pages = True
+    async with aiohttp.ClientSession() as session:
+        while more_pages:
+            async with session.get(url, params=params) as resp:
+                json_resp = await resp.json()
+                playlist_videos.extend(json_resp.get('items'))
+                next_page_token = json_resp.get('nextPageToken')
+                if next_page_token:
+                    params['pageToken'] = next_page_token
+                else:
+                    more_pages = False
+
+        return [
+            YouTubeVideo(
+                video_id=v['snippet']['resourceId']['videoId'],
+                title=v['snippet']['title'],
+                playlist_id=v['snippet']['playlistId'],
+                # JSON response is zero-indexed unlike the UI
+                playlist_position=v['snippet']['position'] + 1,
+            ) for v in playlist_videos
+        ]
+
+
+async def add_youtube_videos_to_store(
+    db: aiosqlite.Connection,
+    videos: List[YouTubeVideo]
+) -> None:
+    for video in videos:
+        await db.execute(
+            (
+                'INSERT INTO youtube_playlist_videos ('
+                '    playlist_id, title, video_id, playlist_position'
+                ')'
```

Comment:
you can drop the columns if you're inserting all of them

Reply:
True, though I've had it drilled into me to be explicit when inserting. will change this as you suggest, and the "I think you can just use video here since it's already a tuple right?" comment above - but together, it just very tightly couples the ordering of properties in the tuple with the schema of the table